server/filestore.go

```diff
@@ -7329,11 +7329,12 @@ func (fs *fileStore) writeFullState() error {
 		binary.MaxVarintLen64 + 8 + 8 // last index + record checksum + full state checksum
 
 	// Do 4k on stack if possible.
-	var raw [4 * 1024]byte
+	const ssz = 4 * 1024
 	var buf []byte
 
-	if sz <= cap(raw) {
-		buf, sz = raw[0:2:cap(raw)], cap(raw)
+	if sz <= ssz {
+		var _buf [ssz]byte
```

Comment:
This seems like it would make runtime work since you are betting it does not grow stack unless on conditional but it would then need to escape the block properly. Being only 4k I think its fine on main stack, I like the make branch.

Reply:
I can put it back to being unconditional, but I did check it with escape analysis and the escape paths are the same (either via `.Seal()` or a realloc with `append`), regardless of whether it's inside the conditional block or not. Just seems like a little way to save the allocation from happening at all if we know from the get-go that we're over 4K.

Reply:
ok if you checked.. Hopefully makes sense though that the runtime seems like it would need to do more work to not decrease stack leaving that block since access to that escapes the block but not the func(). Stack growth is free unless it splits the stack being very big.